src/shogun/preprocessor/RandomFourierGaussPreproc.cpp

```diff
@@ -245,113 +193,87 @@ bool RandomFourierGaussPreproc::init_randomcoefficients() {
+	if (m_dim_feature_space <= 0)
+	{
+		error("ValueError: m_dim_feature_space has value <=0 which isn't allowed!, please set valid value via corresponding setter\n");
+	}
+
+	if (m_kernelwidth <= 0)
+	{
+	error("ValueError: m_kernelwidth has value <=0 which isn't allowed!, please set valid value via corresponding setter\n");
 	}
 
 	io::info("calling CRandomFourierGaussPreproc::init(...)");
 	int32_t num_features =
 	    f->as<DenseFeatures<float64_t>>()->get_num_features();
 
-	if (!test_rfinited()) {
-		dim_input_space = num_features;
-		init_randomcoefficients();
-		ASSERT( test_rfinited())
-	} else {
-		dim_input_space = num_features;
-		// does not reinit if dimension is the same to avoid overriding a previous call of set_randomcoefficients(...)
-		init_randomcoefficients();
+	/* Check if current randweights has the dimensions already
+	NOTE: This also handles the case of rand coefficients not initialized*/
+	is_updated = get_dim_input_space() == num_feats;
+	SG_DEBUG("fit() is_updated = {}",is_updated);
+
+	if (!is_updated)
+	{
+		init_randomcoefficients(num_features);
+		ASSERT(is_updated)
+	}
+	else
+	{
+		// does not reinit if dimension is the same to avoid overriding a
+		// previous call of set_randcoefficients(...)
+		init_randomcoefficients(num_features);
 	}
 }
 
 SGVector<float64_t> RandomFourierGaussPreproc::apply_to_feature_vector(SGVector<float64_t> vector)
 {
-	if (!test_rfinited()) {
-		throw ShogunException(
-				"float64_t * CRandomFourierGaussPreproc::apply_to_feature_vector(...): test_rfinited()==false: you need to call before CRandomFourierGaussPreproc::init (Features *f) OR	1. set_dim_feature_space(const int32 dim), 2. set_dim_input_space(const int32 dim), 3. init_randomcoefficients() or set_randomcoefficients(...) \n");
-	}
+	// if (!test_rfinited()) {
+	// 	throw ShogunException(
+	// 			"float64_t * CRandomFourierGaussPreproc::apply_to_feature_vector(...): test_rfinited()==false: you need to call before CRandomFourierGaussPreproc::init (Features *f) OR	1. set_m_dim_feature_space(const int32 dim), 2. set_dim_input_space(const int32 dim), 3. init_randomcoefficients() or set_randcoefficients(...) \n");
+	// }
 
-	float64_t val = std::sqrt(2.0 / cur_dim_feature_space);
-	SGVector<float64_t> res(cur_dim_feature_space);
+	float64_t val = std::sqrt(2.0 / m_dim_feature_space);
+	SGVector<float64_t> res(m_dim_feature_space);
 
-	for (int32_t od = 0; od < cur_dim_feature_space; ++od) {
-		SGVector<float64_t> wrapper(randomcoeff_multiplicative+od*cur_dim_input_space, cur_dim_input_space, false);
-		res[od] = val * cos(randomcoeff_additive[od] + linalg::dot(vector, wrapper));
+	for (int32_t od = 0; od < m_dim_feature_space; ++od) { //FIXME: Apply Heikos Suggestion
+		SGVector<float64_t> wrapper(m_randomweights+od*dim_input_space, dim_input_space, false);
+		res[od] = val * cos(m_randomoffset[od] + linalg::dot(vector, wrapper));
```

Comment:
pls use a linalg cos that is vectorized

Reply:
or add it

Reply:
Might need to add. Gotta make another PR for that, can add all basic trigonometric in that.